### PR TITLE
refactor: Remove unneeded IsEnabledChangedEventArgs

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by jeromelaban — APPROVED** [review]
It's a change from the original source, but since it's all internal it won't change much. Thanks!

**Comment by MartinZikmund** [comment]
@Youssef1313 If possible, I would prefer to keep the type. Even though it is technically unnecessary, it will make porting from WinUI "harder" as you will always have to do that extra sanity check to ensure, that the method you are overriding is actually the one, that you intend to and not something "slightly different" which is often the case in WinUI C++... I know that this work was for nothing then, but it may really be helpful, to keep comparing WinUI sources to Uno sources as simple as possible (also so that we can later potentially automate it more easily)

**Comment by Youssef1313** [comment]
So should we close this? @jeromelaban @MartinZikmund

File: src/Uno.UI/UI/Xaml/FrameworkElement.cs [hunk 1]
```diff
@@ -740,10 +735,7 @@ private void OnIsEnabledChanged(DependencyPropertyChangedEventArgs args)
 		{
 			UpdateHitTest();
 
-			_isEnabledChangedEventArgs ??= new IsEnabledChangedEventArgs();
-			_isEnabledChangedEventArgs.SourceEvent = args;
-
-			OnIsEnabledChanged(_isEnabledChangedEventArgs);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/FrameworkElement.cs:746]: This part of the code is what led me to this refactoring. It was confusing why we set the needed info to a static `_isEnabledChangedEventArgs` to pass it when we can directly pass it and make the code more readable.

Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/FrameworkElement.cs:746]: The field is not useful. I would suggest changing the `IsEnabledChangedEventArgs` to be a read-only `struct` and just create it and pass it forward. This way we will keep the type for porting parity and avoid potential weird issues coming from it being a static field

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/FrameworkElement.cs:746]: Agreed. Let's keep the code unmodified for source parity, and change the type to a struct.

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/FrameworkElement.cs:746]: @Youssef1313 if changing to a struct does not give much benefit, let's close this PR.

**Comment by Youssef1313** [comment]
Let's close out until it's proven this is problematic to keep source parity for now.

**Pull request closed without merging**